src/error.rs

```diff
@@ -364,22 +367,34 @@ impl Drop for Error {
     }
 }
 
+struct ErrorVTable {
+    source: unsafe fn(*const ()) -> *const (dyn StdError + Send + Sync + 'static),
+    source_mut: unsafe fn(*mut ()) -> *mut (dyn StdError + Send + Sync + 'static),
+}
+
+unsafe fn source_raw<E>(e: *const ()) -> *const (dyn StdError + Send + Sync + 'static)
+where
+    E: StdError + Send + Sync + 'static,
+{
+    e as *const E
+}
+
+unsafe fn source_mut_raw<E>(e: *mut ()) -> *mut (dyn StdError + Send + Sync + 'static)
```

Comment:
I renamed these to object_raw and object_mut_raw since they have nothing to do with the Error::source trait method.
https://doc.rust-lang.org/std/error/trait.Error.html#method.source